lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
-  if (result.isValid === false) {
-    return result;
-  }
+    _.defaultsDeep(fieldSpec, {
+      'mandatory': false,
+      'multivalued': {
+        value: false
+      }
+    });
 
-  _.defaultsDeep(fieldSpec, {
-    'mandatory': false,
-    'multivalued': {
-      value: false
+    if (!fieldSpec.hasOwnProperty('typeOptions')) {
+      fieldSpec.typeOptions = {};
     }
-  });
 
-  if (!fieldSpec.hasOwnProperty('typeOptions')) {
-    fieldSpec.typeOptions = {};
-  }
+    if (!checkAllowedProperties(fieldSpec.typeOptions, this.types[fieldSpec.type].allowedTypeOptions || [])) {
+      throwOrStoreError(`Field ${indexName}.${collectionName}.${fieldName} of type ${fieldSpec.type} is not specified properly`, verboseErrors, errors);
+    }
 
-  if (!checkAllowedProperties(fieldSpec.typeOptions, this.types[fieldSpec.type].allowedTypeOptions || [])) {
-    throwOrStoreError(`Field ${indexName}.${collectionName}.${fieldName} of type ${fieldSpec.type} is not specified properly`, verboseErrors, errors);
-  }
+    returnedTypeOptions = this.types[fieldSpec.type].validateFieldSpecification(fieldSpec.typeOptions);
 
-  returnedTypeOptions = this.types[fieldSpec.type].validateFieldSpecification(fieldSpec.typeOptions);
+    if (typeof returnedTypeOptions === 'boolean') {
+      if (!returnedTypeOptions) {
+        throwOrStoreError(`Field of type ${fieldSpec.type} is not specified properly`, verboseErrors, errors);
+      }
+    }
 
-  if (typeof returnedTypeOptions === 'boolean') {
-    if (!returnedTypeOptions) {
-      throwOrStoreError(`Field of type ${fieldSpec.type} is not specified properly`, verboseErrors, errors);
+    if (errors.length > 0) {
+      return {isValid: false, errors: errors};
     }
-  }
 
-  if (errors.length > 0) {
-    return {isValid: false, errors: errors};
+    fieldSpec.typeOptions = returnedTypeOptions;
+    return {isValid: true, fieldSpec: fieldSpec};
   }
 
-  fieldSpec.typeOptions = returnedTypeOptions;
-  return {isValid: true, fieldSpec: fieldSpec};
-};
-
-/**
- * @param {FieldSpecification} fieldSpec
- * @param {string} indexName
- * @param {string} collectionName
- * @param {string} fieldName
- * @param {boolean} verboseErrors
- * @returns {object}
- * @throws InternalError
- */
-Validation.prototype.curateFieldSpecificationFormat = function validationCurateFieldSpecificationFormat (fieldSpec, indexName, collectionName, fieldName, verboseErrors) {
-  var errors = [];
-
-  verboseErrors = verboseErrors || false;
+  /**
+   * @param {FieldSpecification} fieldSpec
+   * @param {string} indexName
+   * @param {string} collectionName
+   * @param {string} fieldName
+   * @param {boolean} verboseErrors
+   * @returns {object}
+   * @throws InternalError
+   */
+  curateFieldSpecificationFormat(fieldSpec, indexName, collectionName, fieldName, verboseErrors) {
+    var errors = [];
+
+    verboseErrors = verboseErrors || false;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "verboseErrors". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1226)